src/dataflow/sink/avro_ocf.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::fs::File;
+
+use differential_dataflow::hashable::Hashable;
+use log::{debug, error};
+use timely::dataflow::channels::pact::Exchange;
+use timely::dataflow::operators::generic::Operator;
+use timely::dataflow::{Scope, Stream};
+
+use dataflow_types::{AvroOcfSinkConnector, Diff, Timestamp};
+use expr::GlobalId;
+use interchange::avro::{DiffPair, Encoder};
+use repr::{RelationDesc, Row};
+
+pub fn avro_ocf<G>(
+    stream: &Stream<G, (Row, Timestamp, Diff)>,
+    id: GlobalId,
+    connector: AvroOcfSinkConnector,
+    desc: RelationDesc,
+) where
+    G: Scope<Timestamp = Timestamp>,
+{
+    let res = File::create(&connector.path);
```

Comment:
lets move file creation to the planner for now

Reply:
this was not as simple as I thought it would be -- I can't pass around a File object in the AvroOcfSinkConnector because it doesn't satisfy a lot of the asserted trait bounds I think Clone and Eq specifically (not sure what the right verb is in this sentence :) )

Unclear what to do here

Reply:
Yeah, that's intentional, unfortunately. The connector needs to be IPC-able, and file handles cannot cross process boundaries.

Reply:
given the API and constraints we have around files I'm not sure that there's a better thing for us to do; we can definitely try to create the file in the new post planning purify but thats not going to let us remove the call to create() in the dataflow code?